src/Games/NexusMods.Games.StardewValley/Emitters/VersionDiagnosticEmitter.cs

```diff
@@ -48,29 +46,27 @@ public async IAsyncEnumerable<Diagnostic> Diagnose(Loadout.ReadOnly loadout, [En
             yield break;
         }
 
-        var smapiMods = await Helpers
-            .GetAllManifestsAsync(_logger, loadout, _manifestPipeline, onlyEnabledMods: true, cancellationToken)
-            .ToArrayAsync(cancellationToken);
+        var manifests = await Helpers.GetAllManifestsAsync(_logger, loadout.Db, loadout, onlyEnabled: true, _manifestPipeline, cancellationToken);
 
         var apiMods = await _smapiWebApi.GetModDetails(
             os: _os,
             gameVersion,
             smapiVersion,
-            smapiIDs: smapiMods.Select(tuple => tuple.Item2.UniqueID).ToArray()
+            smapiIDs: manifests.Select(tuple => tuple.Item2.UniqueID).ToArray()
         );
 
-        foreach (var tuple in smapiMods)
+        foreach (var tuple in manifests)
         {
-            var (smapiMod, manifest) = tuple;
+            var (manifestLoadoutItem, manifest) = tuple;
 
             var minimumApiVersion = manifest.MinimumApiVersion;
             var minimumGameVersion = manifest.MinimumGameVersion;
 
             if (minimumApiVersion is not null && smapiVersion.IsOlderThan(minimumApiVersion))
             {
                 yield return Diagnostics.CreateSMAPIVersionOlderThanMinimumAPIVersion(
-                    SMAPIMod: smapiMod.AsLoadoutItemGroup().ToReference(loadout),
-                    SMAPIModName: smapiMod.AsLoadoutItemGroup().AsLoadoutItem().Name,
+                    SMAPIMod: manifestLoadoutItem.AsLoadoutFile().AsLoadoutItemWithTargetPath().AsLoadoutItem().Parent.ToReference(loadout),
```

Comment:
This could be a reference to a SMAPI mod submodule though, unless you planned to rebase all modules to the top level with a migration?

Reply:
The diagnostics work fine with data generated by the previous installer.

Reply:
Didn't the previous installer create subgroups for for each smapi module?
Meaning that the name displayed here in that case would be that of the submodule rather than the "top level, collection child, foo, mod" thing's name?

Reply:
I've adapted the diagnostic emitters to work with the new installer without spending too much time on it since we're going to be reworking all of this anyways with #2806.